### ISSUE TITLE
MongoDB default function deleted clang warning

### ISSUE THREAD

**Issue by aleks-f** [pocoproject/poco#4276]
**Describe the bug**
Compile warning

**To Reproduce**
Seen on CI macOS builds

**Expected behavior**
no warning

**Logs**
```
In file included from src/MongoDBTest.cpp:18:
/Users/runner/work/poco/poco/MongoDB/include/Poco/MongoDB/PoolableConnectionFactory.h:125:20: warning: explicitly defaulted move assignment operator is implicitly deleted [-Wdefaulted-function-deleted]
        PooledConnection& operator=(PooledConnection&&) = default;
                          ^
/Users/runner/work/poco/poco/MongoDB/include/Poco/MongoDB/PoolableConnectionFactory.h:128:49: note: move assignment operator of 'PooledConnection' is implicitly deleted because field '_pool' is of reference type 'Poco::ObjectPool<Connection, Connection::Ptr> &' (aka 'ObjectPool<Poco::MongoDB::Connection, SharedPtr<Poco::MongoDB::Connection>> &')
        Poco::ObjectPool<Connection, Connection::Ptr>& _pool;
                                                     ^
```

**Please add relevant environment information:**
 - macOS 
 - clang

**Comment by matejk** [comment]
I had to add additional warning compiler options to cmake to get the aforementioned warning.

That opened a big "can of warms" from all over the Poco source code. Mostly unused parameters and integer comparisons.

**Comment by matejk** [comment]
Fixed in #4316.
